pkg/k8sutil/storage.go

```diff
@@ -0,0 +1,113 @@
+package k8sutil
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// PVSByStorageClass returns a map of persistent volumes using the provided storage class name.
+// returned pvs map is indexed by pv's name.
+func PVSByStorageClass(ctx context.Context, cli kubernetes.Interface, scname string) (map[string]corev1.PersistentVolume, error) {
```

Comment:
wdyt about we have a struct to store the data collect?
would not make it easier to manipulated instead of have these funcs that return a lot of values with maps?

Reply:
I don't see how this would bring us any advantage tbh. A map from name to value is adequate to what we are trying to achieve.

Reply:
By looking here I thought that could be something like 
load data and store in a temp struct and then after we could check out 

However, that would not be a blocker and is more a matter of preference. So. I am closing this one.